ui/src/app/views/home-page/component.ts

```diff
@@ -43,26 +45,33 @@ export class HomePageComponent implements OnInit {
   private workspaceResearchPurposeFilter = new WorkspaceResearchPurposeFilter();
   private workspaceNameComparator = new WorkspaceNameComparator();
 
+  errorText: string;
   workspaceList: WorkspaceResponse[] = [];
   workspacesLoading = false;
   workspaceAccessLevel = WorkspaceAccessLevel;
   constructor(
+      private errorHandlingService: ErrorHandlingService,
       private route: ActivatedRoute,
       private router: Router,
       private workspacesService: WorkspacesService,
   ) {}
   ngOnInit(): void {
+    this.errorText = null;
     this.workspacesLoading = true;
     this.workspacesService.getWorkspaces()
-        .subscribe(
-            workspacesReceived => {
-              this.workspaceList = workspacesReceived.items;
-              this.workspacesLoading = false;
-            },
-            error => {
-              // TODO: Add Error Message.
-              this.workspacesLoading = false;
-            });
+      .subscribe(
+        workspacesReceived => {
+          this.workspaceList = workspacesReceived.items;
+          this.workspacesLoading = false;
+        },
+        error => {
+          // if loading workspaces throws an error, display to the user
+          const response: ErrorResponse = ErrorHandlingService.convertAPIError(error);
+          if (response.message !== null) {
+            this.errorText = response.message;
```

Comment:
The only way to unset this is to refresh the page, right - the error state is terminal?

Reply:
errorText is now cleared if it's not a MailChimp error